### PR TITLE
[12_5_X][L1T] Backport of: Phase-2, correlator developments

### PR DESCRIPTION
**Pull request by cecilecaillol**
#### PR description:

This PR includes the latest devlopments to the Phase-2 L1 correlator. The local PR is https://github.com/cms-l1t-offline/cmssw/pull/1031

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

Backport of https://github.com/cms-sw/cmssw/pull/39397. We need it for our Phase-2 MC production.

### CODE REVIEW

**Comment by cecilecaillol** [comment]
backport of https://github.com/cms-sw/cmssw/pull/39397

**Comment by cecilecaillol** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by AdrianoDee** [comment]
@cecilecaillol could you apply the same changes applied to the PR to `master`?

**Comment by cecilecaillol** [comment]
please test

**Comment by AdrianoDee** [comment]
+upgrade
- backport of https://github.com/cms-sw/cmssw/pull/39397;
- needed for 12_5_X L1T MC production.

**Comment by cecilecaillol** [comment]
+l1

**Review by perrotta — COMMENTED** [review]
This BuildFile was cleaned in the master: please compare with the same BuildFile in the master and remove what not needed (in particular the duplicate) and add what missing (if any)

**Comment by cecilecaillol** [comment]
please test

File: L1Trigger/Phase2L1ParticleFlow/BuildFile.xml [hunk 1]
```diff
@@ -10,6 +10,7 @@
 <use name="L1Trigger/DemonstratorTools"/>
 <use name="CommonTools/Utils"/>
 <use name="CommonTools/MVAUtils"/>
+<use name="L1Trigger/DemonstratorTools"/>
```
[the remark this answers is not in the document]
Comment (cecilecaillol) [reply L1Trigger/Phase2L1ParticleFlow/BuildFile.xml:11]: Fixed

**Comment by cecilecaillol** [comment]
+l1

**Comment by perrotta** [comment]
+1

**Comment by perrotta** [comment]
ping bot

**Comment by perrotta** [comment]
ping bot
(once again...)

**Comment by perrotta** [comment]
ping bot
(once again and again...)

**Comment by AdrianoDee** [comment]
+upgrade
Re-sign

**Pull request merged**